### DISCUSSION TITLE
CenterCrop after Resize with the same size value

### DISCUSSION THREAD

**Discussion by tolgacangoz** [huggingface/diffusers#7042] [General]
Hi `diffusers` community,
I wonder what is the reason for using `transforms.CenterCrop(size)` after `transforms.Resize(size)` with the same values. Isn't CenterCrop(size) unnecessary? [Example](https://github.com/huggingface/diffusers/blob/main/examples/controlnet/train_controlnet.py#L671).

**Comment by asomoza** [comment]
`size` is an `int` in that script:

```python
parser.add_argument(
        "--resolution",
        type=int,
        default=512,
        help=(
            "The resolution for input images, all the images in the train/validation dataset will be resized to this"
            " resolution"
        ),
    )
```

transforms.Resize(args.resolution)

> If size is an int, smaller edge of the image will be matched to this number. i.e, if height > width, then image will be rescaled to (size * height / width, size).

transforms.CenterCrop(args.resolution)

> If size is an int instead of sequence like (h, w), a square crop (size, size) is made.

So in short, Is necessary if the image is not a square image, resize maintains the aspect ratio of the image.

**Reply by tolgacangoz** [reply]
Hmm, you are right, thank you 👍.
